src/tactic/core.lean

```diff
@@ -107,6 +107,23 @@ meta def kreplace (e old new : expr) (md := semireducible) (unify := tt)
 
 end expr
 
+namespace name
+
+/-- Auxiliary function for `name.contains_sorry`. -/
```

Comment:
Do I understand the purpose of this definition correctly?
```suggestion
/-- Does `sorry` occur in the declaration `nm` or any of the declarations
of the form `pre.__proof_i` used by `nm`?

Auxiliary function for `name.contains_sorry`. -/
```

Reply:
Yes, basically. I don't explicitly check that the recursive names end in `_proof_i` (which is unlikely to result in false positives). I added a note to this extend in the docstring of `contains_sorry`.